### ISSUE TITLE
Fuzzer-identified panics

### ISSUE THREAD

**Issue by sigaloid** [nickbabcock/boxcars#162]
Hi, using a fuzzer with better features identified a number of crashes:

[crashes.zip](https://github.com/user-attachments/files/16185879/crashes.zip)

```
thread 'main' panicked at /home/mattop/boxcars/src/network/mod.rs:202:25:
attempt to subtract with overflow
```
\- OR -
```
thread 'main' panicked at /home/mattop/boxcars/src/network/mod.rs:185:23:
attempt to add with overflow
```

In order to be resistant to these crashes, it might be a good idea to use checked math like referenced here: https://github.com/nickbabcock/boxcars/issues/11#issuecomment-358477316

**Comment by nickbabcock** [comment]
Thank you for fuzzing. I can confirm that these test cases cause overflows.

I'm about to be on holiday for the next couple weeks, so you're more than welcome to add in checked math to these places.

I'm disappointed that the existing fuzzing suite (ie: the libfuzzer `no-crc-body` suite) didn't catch these issues. I would be interested in hearing your thoughts on improving the status quo.

**Comment by sigaloid** [comment]
Ah, interesting that cargo fuzz found it as well! I also tried it, and found nothing for a while, so I added [AFL++](https://aflplus.plus/) which ended up finding it quickly whereas cargo fuzz didn't.

Anyway, submitted #163 to fix this! Confirmed it fixed these panics.

AFL++ tends to be better in a lot of cases, however, especially given a large corpus for relevant inputs. I was able to reuse the various test artifacts from the existing `fuzz/` and `assets/replays/` folders.
